### PR TITLE
fix: Overall capacity summary is inconsistent with owner workloads #282

### PR DESCRIPTION
**Pull request by YuliaMittova**
PR description: Bugfix for issue #282

### CODE REVIEW

File: src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java [hunk 1]
```diff
@@ -215,30 +214,22 @@ private PlanningTrendsResponse.PlanningAccuracyTrend calculatePlanningAccuracyTr
     }
 
     private CapacityPlanningResponse.OverallCapacity calculateOverallCapacity(List<Feature> features) {
+        final int maxCapacityPerPerson = 10; // As per simplified implementation
```
Comment (innokenty) [inline src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java]: I seeee, so the capacity is just hardcoded to be 10 in the AI implementation.... Okay, we can live with this. 

But re-introducing a magic constant here is a terrible code style, I extracted it into a proper constant on class level and going to force-push this PR with this change.

**Review by innokenty — APPROVED** [review]

File: src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java [hunk 2]
```diff
@@ -215,30 +216,22 @@ private PlanningTrendsResponse.PlanningAccuracyTrend calculatePlanningAccuracyTr
...
-                            .count();
-                    return assigned > 0 ? (double) completed / assigned * 100 : 0;
-                })
-                .average()
-                .orElse(0.0);
-
-        double utilizationRate = round(totalUtilization, 1);
+                .mapToInt(List::size)
+                .sum();
+        double totalUtilization = (double) totalAmountOfFeatures / (DEFAULT_CAPACITY_PER_PERSON * totalResources);
```
Comment (ElizavetaP) [inline src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java:230]: This bugfix introduces a small edge-case regression: when there are no owners, the calculation becomes 0 / 0, which produces NaN and then fails during rounding.
It is not the main issue, but it would be better to address it as well. 
A simple way to fix this would be to add an explicit zero-owner check, for example as in the golden implementation.

**Review by ElizavetaP — CHANGES_REQUESTED** [review]

File: src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java (hunk 2 above)
Comment (innokenty) [reply src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java:230]: I am not sure if maybe we never end here, because there's a safeguard somewhere before, but please add a test – that's a good use case!

Comment (YuliaMittova) [reply src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java:230]: @ElizavetaP I tried to apply golden implementation from here and I see that it calculates totalUtilization with a mistake:

```
        double totalUtilization = features.stream()
                .filter(f -> f.getFeatureOwner() != null)
                .collect(Collectors.groupingBy(Feature::getFeatureOwner))
                .values()
                .stream()
                .mapToDouble(ownerFeatures -> {
                    int assigned = ownerFeatures.size();
                    int completed = (int) ownerFeatures.stream()
                            .filter(f -> f.getPlanningStatus() == FeaturePlanningStatus.DONE)
                            .count();
                    return assigned > 0 ? (double) completed / assigned * 100 : 0;
                })
                .average()
                .orElse(0.0);
```
For example, if one person has 10 tasks in progress and another one has 1 task in progress then totalUtilization will be 0 because none of the tasks are completed. Can you tell what is expected here? Is this logic intended?

Comment (ElizavetaP) [reply src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java:230]: @YuliaMittova I only meant the NaN edge case.

If there are no owners, this becomes 0 / 0, which produces NaN. My comment was only about guarding that case, not about copying the whole golden calculation logic.

Comment (innokenty) [reply src/main/java/com/sivalabs/ft/features/domain/PlanningAnalyticsService.java:230]: That's true, I'm already fixing this.

**Review by innokenty — APPROVED** [review]
I now fixed this dataset too:

- Bug fix: division by zero when no owners exist
  `calculateOverallCapacity` crashed with `NaN` when all features had `null` owner
  (or the database was empty), because it divided by `totalResources` without
  a zero check. Added an early return for this case.

- Refactor: unified duplicate stream operations
  `calculateOverallCapacity` was streaming and filtering features three separate times
  to compute owners, feature count, and overallocated resources. Consolidated into
  a single `groupingBy` call, reusing the resulting map for all three computations.

- Test: added edge-case coverage for zero-owner scenarios
  - Empty database — no features, no releases
  - All features have `null` owner — 2 features exist but none are assigned

- Test: added `workloadByOwner` assertions
  All four capacity tests previously only checked `workloadByOwner` was not null.
  Added per-owner assertions (`owner`, `currentWorkload`, `capacity`) using
  order-independent matching, so the tests now verify consistency between
  `overallCapacity` and `workloadByOwner` — the core of the original bug.

**Review by ElizavetaP — APPROVED** [review]
